src/main/java/com/impinjCtrl/ReaderController.java

```diff
@@ -12,26 +12,39 @@
 import lib.PropertyUtils;
 import okhttp3.*;
 import org.json.simple.JSONObject;
+//import org.json.simple.JSONArray;
 import org.json.simple.parser.JSONParser;
 
 import javax.net.ssl.*;
+import java.io.FileWriter;
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.security.KeyManagementException;
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
 import java.util.Scanner;
+import com.sun.istack.internal.Nullable;
 
 public class ReaderController {
 
     public static final String EVENT_START_READER = "startreader";
     public static final String EVENT_TERMINATE_READER = "terminatereader";
     public static final String EVENT_TRANSFER_DATA = "rxdata";
+    public static final String EVENT_TRANSFER_DATA_TESTMODE = "rxdatatest";
     public static final String EVENT_GET_READER_STATUS = "getreaderstatus";
     public static final String EVENT_READER_STATUS = "readerstatus";
     public static String mEventId;
     public static String mSocketId;
+    public static String mRaceId;
+    public static Long mValidIntervalMs;
+    public static JSONObject mSlaveEpcMap;
+    public static String mLogFileName;
+    public static JSONObject mReadWrapper;
+    public static JSONObject mRecordsHashTable;
+    public static JSONObject mSlaveEpcStat;
+    //public static JSONArray mReadResultRaw;
+    public static HttpClient mHttpClient;
```

Comment:
it seems not so suitable to let all of these to become static, each one can be a leak.

Reply:
if really want these, make sure that they will be cleaned or destroy.